sonic-net/sonic-buildimage
# Implemented secure upgrade

#### What I did
Implemented secure upgrade in Sonic. 

How I did it
During sonic_installer install, added secure upgrade image verification. 
HLD can be found in the following PR: https://github.com/sonic-net/SONiC/pull/1024
#### Why I did it
Feature is used to allow image was not modified since built from vendor. During installation, image can be verified with a signature attached to it.
#### How I did it
Feature includes image signing during build (in sonic buildimage repo) and verification during image install (in sonic-utilities).
#### How to verify it
In order for image verification - image must be signed - need to provide signing key and certificate (paths in SECURE_UPGRADE_DEV_SIGNING_KEY and SECURE_UPGRADE_DEV_SIGNING_CERT in rules/config) during build , and during image install, need to enable secure boot flag in bios, and signing_certificate should be available in bios.
#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Link to config_db schema for YANG module changes


#### A picture of a cute
![b825dc7667d17258f5d1955f5c94d88c--github-logo-website-logo](https://user-images.githubusercontent.com/99744138/184639624-47ac744d-a453-419c-96b4-f192c3d72c02.jpg)
 animal (not mandatory but encouraged)
